webhooks-with-pgstream.mdx

```diff
@@ -0,0 +1,555 @@
+---
+title: 'Postgres webhooks with pgstream'
+description: 'A simple tutorial for calling webhooks on Postgres data and schema changes using pgstream.'
+image:
+  src: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/pgstream/introducing-pgstream@2x.jpg
+  alt: pgstream
+author: Tudor Golubenco
+authorEmail: tudor@xata.io
+date: 08-27-2024
+tags: ['postgres', 'fpPgstream', 'oss', 'open-source']
+published: true
+slug: postgres-webhooks-with-pgstream
+ogImage: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/pgstream/introducing-pgstream@2x.jpg
+---
+
+[pgstream](https://github.com/xataio/pgstream) is a CDC (Change-Data-Capture) tool focused on PostgreSQL. Among other [things](https://github.com/xataio/pgstream?tab=readme-ov-file#features), it can be used to call webhooks whenever there is a data (or schema) change in a Postgres database. This means that whenever a row is inserted, updated, or deleted, a webhook is called with the data from the row.
+
+In this article we're going to dive in the pgstream webhooks functionality, including more advanced topics, like sending the old and new values, TOAST support, and schema changes support.
+
+![Postgres webhooks with pgstream](/images/pgstream-webhooks.png)
+
+## Preparation
+
+For this tutorial we need a Postgres instance that has logical replication enabled and the [wal2json](https://github.com/eulerto/wal2json) output plugin loaded. There are a number of ways to do this, but we're going to use the [docker-compose file](https://github.com/xataio/pgstream/blob/main/build/docker/docker-compose.yml) from the pgstream repo:
+
+```sh
+git clone https://github.com/xataio/pgstream.git
+```
+
+And then start the Postgres instance:
+
+```sh
+docker compose -f build/docker/docker-compose.yml up db
+```
+
+Then, in a new terminal window, you can connect to it and create a simple table to play with:
+
+```sh
+psql "postgres://postgres:postgres@localhost:5432/postgres"
+```
+
+At the psql prompt:
+
+```sql
+CREATE TABLE users(id SERIAL PRIMARY KEY, name TEXT, role TEXT);
+```
+
+Let's now install `pgstream`. One option is to build it from source, since we already have the repo cloned:
```

Comment:
We can also use `go install github.com/xataio/pgstream@latest`.
Maybe we can link to the readme here:
https://github.com/xataio/pgstream/tree/main?tab=readme-ov-file#installation

Reply:
Thanks, replaced the homebrew part with a link to the installation instructions.